src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -830,6 +830,19 @@ private void ParseProject(string firstLine)
                         line = ReadLine();
                     }
                 }
+                else if (line.StartsWith("Project(", StringComparison.Ordinal))
+                {
+                    // Another Project spotted instead of EndProject for the current one - solution file is malformed
+                    string warning = ResourceUtilities.FormatResourceStringStripCodeAndKeyword(out _, out _, "Shared.InvalidProjectFile",
+                        _solutionFile, proj.ProjectName);
+                    SolutionParserWarnings.Add(warning);
```

Comment:
Because of this check

https://github.com/dotnet/msbuild/blob/5b8d459b87b823ebd185d3fddfa3aa4eb415081d/src/Build/Graph/GraphBuilder.cs#L265-L273

this will fail graph builds like this:

```sh-session
msbuild on fix-skipping-project-without-EndProject [$!] •NET v5.0.100-rc.1.20452.10
❯ artifacts\bin\bootstrap\net472\MSBuild\Current\Bin\MSBuild.exe .\MSBuild.sln -t:lkejt -graph
Microsoft (R) Build Engine version 16.9.0-dev-20516-01+9ca2a18c3 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error : MSB4262: Solution file ".\MSBuild.sln" contains the following warnings and errors:
MSBUILD : error :            Warnings: The project file is malformed: "S:\msbuild\MSBuild.sln". Solution Items;The project file is malformed: "S:\msbuild\MSBuild.sln". Samples;The project file is malformed: "S:\msbuild\MSBuild.sln". Microsoft.Build
MSBUILD : error :            Errors:
MSBUILD : error :

Build FAILED.

MSBUILD : error : MSB4262: Solution file ".\MSBuild.sln" contains the following warnings and errors:
MSBUILD : error :            Warnings: The project file is malformed: "S:\msbuild\MSBuild.sln". Solution Items;The project file is malformed: "S:\msbuild\MSBuild.sln". Samples;The project file is malformed: "S:\msbuild\MSBuild.sln". Microsoft.Build
MSBUILD : error :            Errors:
MSBUILD : error :
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.06
```

I'm not sure what the right thing to do here is. It doesn't really make sense for a graph build to fail because of a solution error when a normal build wouldn't fail. But that appears to be the standard behavior for graph builds, and the only use of `SolutionParserWarnings` that I actually see in the codebase. So this doesn't seem to make it worse. Maybe we should leave it?

Reply:
I've added this warning to make it clear for user that solution file is malformed.
On the other hand building with Visual Studio doesn't return any warning for malformed solution file, so it should be further discussed whether to warn user about malformed solution or remove warning to keep the graph build working.

Reply:
@cdmihai it looks like graph mode is the only thing that looks at `SolutionParserWarnings`. Should we stop having it fail if there is one? Then we could leave this (reasonable IMO) new warning here . . .

Reply:
It does sound pretty reasonable to change the GraphBuilder to:
- Log the solution parsing warnings but continue the build instead of stopping it.
- Continue stopping the build on solution errors.

Reply:
@cdmihai Should the suggested change be part of this pull request, or will there be another issue created for the GraphBuilder?

Reply:
Let's change the graph stuff separately.

Reply:
If you're fine with doing the change, I'd go with adding it to this PR as a separate commit which does not get squashed. Otherwise, please open an issue.

Reply:
Concurrent post. I defer to @rainersigwald :)